### PR TITLE
[LA64_DYNAREC] Added ROUNDSD and CVTDQ2PD opcodes

### PR DESCRIPTION
**Pull request by xiangzhai**
Hi,

Please review my patch.

Thanks,
Leslie Zhai

### CODE REVIEW

File: src/dynarec/la64/dynarec_la64_660f.c [hunk 1]
```diff
@@ -528,6 +528,32 @@ uintptr_t dynarec64_660F(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int
...
             switch (opcode) {
+                case 0x0B:
+                    INST_NAME("ROUNDSD Gx, Ex, Ib");
+                    nextop = F8;
+                    GETGX(q0, 1);
+                    GETEXSD(q1, 0, 1);
+                    u8 = F8;
+                    v0 = fpu_get_scratch(dyn);
+                    v1 = fpu_get_scratch(dyn);
+                    XVREPLVE0_D(v0, q1);
```
Comment (ksco) [inline src/dynarec/la64/dynarec_la64_660f.c:539]: What is this for? I don't not follow.

File: src/dynarec/la64/dynarec_la64_660f.c [hunk 2]
```diff
@@ -528,6 +528,32 @@ uintptr_t dynarec64_660F(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int
...
+                        x87_restoreround(dyn, ninst, u8);
+                    } else if ((u8 & 0x3) == 0x0) {
+                        VFRINTRNE_D(v1, v0);
+                    } else if ((u8 & 0x3) == 0x1) {
+                        VFRINTRM_D(v1, v0);
+                    } else if ((u8 & 0x3) == 0x2) {
+                        VFRINTRP_D(v1, v0);
+                    } else if ((u8 & 0x3) == 0x3) {
+                        VFRINTRZ_D(v1, v0);
+                    }
```
Comment (ksco) [inline src/dynarec/la64/dynarec_la64_660f.c:552]: Use an array `round_round` to simplify this.

File: src/dynarec/la64/dynarec_la64_660f.c [hunk 3]
```diff
@@ -528,6 +528,32 @@ uintptr_t dynarec64_660F(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int
...
+                    XVINSVE0_D(q0, v1, 0);
+                    XVINSGR2VR_D(q0, xZR, 2);
+                    XVINSGR2VR_D(q0, xZR, 3);
```
Comment (ksco) [inline src/dynarec/la64/dynarec_la64_660f.c:555]: I don't understand this either, please explain it.

**Review by ksco — CHANGES_REQUESTED** [review]
Some part seems very confusing, need some explainations.

File: src/dynarec/la64/dynarec_la64_660f.c (hunk 1 above)
Comment (xiangzhai) [reply src/dynarec/la64/dynarec_la64_660f.c]: copy src lo64 to tmp every elements.

File: src/dynarec/la64/dynarec_la64_660f.c (hunk 3 above)
Comment (xiangzhai) [reply src/dynarec/la64/dynarec_la64_660f.c]: set dest hi128 to zero.
please pointing out my fault.

File: src/dynarec/la64/dynarec_la64_660f.c (hunk 1 above)
Comment (ksco) [reply src/dynarec/la64/dynarec_la64_660f.c]: Yeah I know what this instruction do, but what is it for?

File: src/dynarec/la64/dynarec_la64_660f.c [hunk 4]
```diff
@@ -49,6 +49,10 @@ uintptr_t dynarec64_660F(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int
     MAYUSE(eb1);
     MAYUSE(eb2);
     MAYUSE(j64);
+    #if STEP > 1
+    static const int8_t round_round[] = { 0xE, 0x2, 0x6, 0xA};
```
Comment (ksco) [inline src/dynarec/la64/dynarec_la64_660f.c:53]: Please make this more readable.

**Review by ksco — APPROVED** [review]
Hmmm I would prefer some self explanatory macros, but fine.

**Comment by xiangzhai** [comment]
Thanks :beers:

**Pull request merged**